src/controllers/controllerhidreporttabsmanager.cpp

```diff
@@ -0,0 +1,498 @@
+        pJsProxy->sendOutputReport(reportId, reportData);
+    }
+}
+
+void ControllerHidReportTabsManager::populateHidReportTable(
+        QTableWidget* pTable,
+        const hid::reportDescriptor::Report& pReport,
+        hid::reportDescriptor::HidReportType reportType) {
+    // Temporarily disable updates to speed up populating
+    pTable->setUpdatesEnabled(false);
+
+    // Reserve rows up-front
+    const auto& controls = pReport.getControls();
+    pTable->setRowCount(static_cast<int>(controls.size()));
+
+    // Set the delegate once if needed
+    if (reportType != hid::reportDescriptor::HidReportType::Input) {
+        pTable->setItemDelegateForColumn(5, new ValueItemDelegate(pTable));
+    }
+
+    bool showVolatileColumn = (reportType == hid::reportDescriptor::HidReportType::Feature ||
+            reportType == hid::reportDescriptor::HidReportType::Output);
+
+    // Set headers
+    QStringList headers = {QStringLiteral("Byte Position"),
+            QStringLiteral("Bit Position"),
+            QStringLiteral("Bit Size"),
+            QStringLiteral("Logical Min"),
+            QStringLiteral("Logical Max"),
+            QStringLiteral("Value"),
+            QStringLiteral("Physical Min"),
+            QStringLiteral("Physical Max"),
+            QStringLiteral("Unit Scaling"),
+            QStringLiteral("Unit"),
+            QStringLiteral("Abs/Rel"),
+            QStringLiteral("Wrap"),
+            QStringLiteral("Linear"),
+            QStringLiteral("Preferred"),
+            QStringLiteral("Null")};
+    if (showVolatileColumn) {
+        headers << QStringLiteral("Volatile");
+    }
+    headers << QStringLiteral("Usage Page") << QStringLiteral("Usage");
+
+    pTable->setColumnCount(headers.size());
+    pTable->setHorizontalHeaderLabels(headers);
+    pTable->verticalHeader()->setVisible(false);
+
+    // Helpers
+    auto createReadOnlyItem = [](const QString& text, bool rightAlign = false) {
+        auto* item = new QTableWidgetItem(text);
+        item->setFlags(item->flags() & ~Qt::ItemIsEditable);
+        if (rightAlign) {
+            item->setTextAlignment(Qt::AlignRight | Qt::AlignVCenter);
+        }
+        return item;
+    };
+    auto createValueItem = [reportType](int minVal, int maxVal) {
+        auto* item = new QTableWidgetItem;
+        QFont font = item->font();
+        font.setBold(true);
+        item->setFont(font);
+        if (reportType == hid::reportDescriptor::HidReportType::Input) {
+            item->setFlags(item->flags() & ~Qt::ItemIsEditable);
+        } else {
+            item->setFlags(item->flags() | Qt::ItemIsEditable);
+            item->setData(Qt::UserRole, QVariant::fromValue(QPair<int, int>(minVal, maxVal)));
+        }
+        return item;
+    };
+
+    int row = 0;
+    for (const auto& pControl : controls) {
+        // Column 0 - Byte Position
+        auto* bytePositionItem = createReadOnlyItem(QStringLiteral("0x%1").arg(QString::number(
```

Comment:
The type of auto cant be guessed here
```suggestion
        QTableWidgetItem* bytePositionItem = createReadOnlyItem(QStringLiteral("0x%1").arg(QString::number(
```